Gems/ScriptCanvas/Code/Editor/View/Windows/MainWindow.cpp

```diff
@@ -4246,13 +4261,61 @@ namespace ScriptCanvasEditor
             RemoveSystemTickAction(SystemTickActionFlag::CloseNextTabAction);
             CloseNextTab();
         }
+    }
 
-        if (m_systemTickActions == 0)
+    void MainWindow::ClearSaveAttempt()
+    {
+        m_saveAttemptInProgress = false;
+
+        if (!m_systemTickActions)
         {
             AZ::SystemTickBus::Handler::BusDisconnect();
         }
     }
 
+    bool MainWindow::EvaluateSaveAttempt()
+    {
+        const AZ::s64 k_saveAttemptSeconds = 20;
+
+        if (m_saveAttemptInProgress)
+        {
+            auto saveDuration = AZStd::chrono::seconds(AZStd::chrono::system_clock::now() - m_saveAttemptTime).count();
+
+            if (saveDuration > k_saveAttemptSeconds)
+            {
+                WarnOnFailedSaveAttempt();
+            }
```

Comment:
Shouldn't the `m_saveAttemptInProgress` be reset to false here?

Reply:
no, the editor will be closed after this, and then constructed anew if re-opened